lib/queueservice.js

```diff
@@ -0,0 +1,82 @@
+import xml2js from 'xml2js';
+import Debug from 'debug';
+import request from 'superagent-promise';
+
+let debug = Debug('taskcluster-docker-worker:queueService');
+
+export default class TaskQueue {
+  constructor(config) {
+    this.queues;
+    this.workerType = config.workerType;
+    this.provisionerId = config.provisionerId;
+    this.client = config.queue;
+    this.log = config.log;
+  }
+
+  async getQueues() {
+    if (!this.queues || (new Date(this.queues.expires)).getTime() > Date.now()) {
+      this.queues = await this.client.pollTaskUrls(this.provisionerId, this.workerType);
+    }
+    return this.queues.queues;
+  }
+
+  async getTasksFromQueue(queue, numberOfTasks) {
+    let tasks = [];
+    let uri = `${queue.signedPollUrl}&numofmessages=${numberOfTasks}`;
+    debug(`requesting: ${uri}`);
+    let response = await request
+      .get(uri)
+      .buffer()
+      .end();
+
+    let xml = await new Promise((accept, reject) => {
+      xml2js.parseString(response.text, (err, json) => {
+        err ? reject(err) : accept(json)
+      });
+    });
+
+    if(!xml.QueueMessagesList) { return tasks }
+
+    for(let message of xml.QueueMessagesList.QueueMessage) {
+      let payload = new Buffer(message.MessageText[0], 'base64').toString();
+      let task = JSON.parse(payload);
```

Comment:
I know that the azure queue contains messages (so that name is taken), but can please not call these for_tasks_.
Maybe advice, and wrap them in a class so we can make`deleteTaskFromQueue` into a method`..
```
Class Advice {
  constructor(payload, deleteUrl) {...}
  async remove() {try {await request.del(this.deleteUri)....} }
}
```

The azure queues contains advice or hints that a task _may_ be be pending. It might also not be pending or even exist. Hence calling it _advice_ might work well.